lib/Doctrine/DBAL/Schema/SQLServerSchemaManager.php

```diff
@@ -37,6 +37,31 @@ class SQLServerSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function dropDatabase($database)
+    {
+        try {
+            parent::dropDatabase($database);
+        } catch (DBALException $exception) {
+            /** @var \Doctrine\DBAL\Driver\DriverException $exception */
+            $exception = $exception->getPrevious();
+
+            // If we have a error code 3702, the drop database operation failed
+            // because of active connections on the database.
+            // To force dropping the database, we first have to close all active connections
+            // on that database and issue the drop database operation again.
+            if ($exception->getErrorCode() !== 3702) {
+                throw $exception;
+            }
+
+            $this->_execSql($this->_platform->getCloseActiveDatabaseConnectionsSQL($database));
```

Comment:
@deeky666 one possible solution is to turn the return type into a `string[]`, then have a base method always returning an empty `array`. `array_walk`-ing over the result is sufficient to ensure consistency then.